File: src/main/java/org/gecko/tools/StateCreatorTool.java
package org.gecko.tools;

import javafx.scene.Node;
import org.gecko.view.views.viewelement.EdgeViewElement;
import org.gecko.view.views.viewelement.RegionViewElement;
import org.gecko.view.views.viewelement.StateViewElement;
import org.gecko.view.views.viewelement.SystemConnectionViewElement;
import org.gecko.view.views.viewelement.SystemViewElement;
import org.gecko.view.views.viewelement.VariableBlockViewElement;

public class StateCreatorTool extends Tool {

    private static final String NAME = "State Creator";

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String getIconPath() {
        //TODO stub
        return null;
    }

    @Override
    public void visitView(Node view) {
        //TODO stub
    }

    @Override
    public void visit(StateViewElement stateViewElement) {

    }

    @Override
    public void visit(EdgeViewElement edgeViewElement) {

    }

    @Override
    public void visit(RegionViewElement regionViewElement) {

    }

    @Override
    public void visit(SystemViewElement systemViewElement) {

    }

    @Override
    public void visit(SystemConnectionViewElement systemConnectionViewElement) {

    }

    @Override
    public void visit(VariableBlockViewElement variableBlockViewElement) {

    }
}

File: src/main/kotlin/org/gecko/tools/StateCreatorTool.kt
package org.gecko.tools

import javafx.event.EventHandler
import javafx.scene.input.MouseButton
import javafx.scene.input.MouseEvent
import org.gecko.actions.Action
import org.gecko.actions.ActionManager
import org.gecko.view.views.ViewElementPane

/**
 * A concrete representation of a state-creating-[Tool], utilized for creating a
 * [StateViewModel][org.gecko.viewmodel.StateViewModel].
 */
class StateCreatorTool(actionManager: ActionManager) : Tool(actionManager, ToolType.STATE_CREATOR, false) {
    override fun visitView(pane: ViewElementPane) {
        super.visitView(pane)
        pane.draw().onMouseClicked = EventHandler<MouseEvent> { event: MouseEvent ->
            if (event.button != MouseButton.PRIMARY) {
                return@EventHandler
            }
            val position = pane.screenToWorldCoordinates(event.screenX, event.screenY)
            val createStateAction: Action =
                actionManager.actionFactory.createState(position)
            actionManager.run(createStateAction)
        }
    }
}
